packages/next/src/server/app-render/app-render.tsx

```diff
@@ -512,6 +512,22 @@ async function generateDynamicFlightRenderResult(
       onError,
     }
   )
+  await waitAtLeastOneReactRenderTask()
+
+  if (
+    ctx.staticGenerationStore.pendingRevalidates ||
+    ctx.staticGenerationStore.revalidatedTags ||
+    ctx.staticGenerationStore.pendingRevalidateWrites
+  ) {
+    const promises = Promise.all([
+      ctx.staticGenerationStore.incrementalCache?.revalidateTag(
+        ctx.staticGenerationStore.revalidatedTags || []
+      ),
+      ...Object.values(ctx.staticGenerationStore.pendingRevalidates || {}),
+      ...(ctx.staticGenerationStore.pendingRevalidateWrites || []),
+    ])
+    ctx.renderOpts.waitUntil = (p) => promises.then(() => p)
```

Comment:
this is not correct, if anything this should be assigning to `renderOpts.pendingWaitUntil`. the `waitUntil` function is a different thing

Reply:
But the `renderOpts.pendingWaitUntil` property is not available on this type. It is defined on the type `StaticGenerationContext` and `ctx` is a different type here.

Reply:
that's true, and i'd need to double check why exactly those types are different here. but what i'm saying is that this line does not do what it's intended to do, i.e. these promises will not be awaited by anything.

Reply:
Oh I get what you mean! `FlightRenderResult` does not pass the `waitUntil` function to its superclass `RenderResult`.. will raise a PR addressing this.

Reply:
I have raised a PR to fix this issue: https://github.com/vercel/next.js/pull/70719

Reply:
thanks, but actually i'm not sure if we're not going to go with a different approach (https://github.com/vercel/next.js/pull/70686#discussion_r1783888770). so we'll see

Reply:
oh ok, so does this mean that `revalidateTag` won't be allowed in server components? if so, I'll just close this new PR I raised.

Reply:
closed my PR.